### PR TITLE
Drive with joystick input

### PR DESCRIPTION
**Pull request by minhnguyenbhs**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -10,6 +22,216 @@ public static enum Mode {
...
+        public static final int rightPivotMotorID = 13;
+
+        public static final int pivotEncoderID = 0;
+
+        public static final int telescopeMotorID = 18;
+
+        public static final int wristMotorID = 17;
+
+        public static final int rightIntakeMotorID = 15;
+        public static final int leftIntakeMotorID = 16;
```
Comment (jkleiber) [inline src/main/java/frc/robot/Constants.java:57]: These shouldn't be in this PR

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -10,6 +22,216 @@ public static enum Mode {
...
+
+    }
+
+    public static final class DIOPorts {
+
+        public static final int switch1 = 1;
+        public static final int switch2 = 2;
     }
```
Comment (jkleiber) [inline src/main/java/frc/robot/Constants.java:65]: I don't think this is relevant to this PR

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -10,6 +22,216 @@ public static enum Mode {
...
+
+        public static final double poseMoveTranslationkP = 1;
+        public static final double poseMoveTranslationMaxVel = 3;
+        public static final double poseMoveTranslationMaxAccel = 3;
+
+        public static final double poseMoveRotationkP = 0.05;
+        public static final double poseMoveRotationMaxVel = Math.PI;
+        public static final double poseMoveRotationMaxAccel = Math.PI;
+
+    }
```
Comment (jkleiber) [inline src/main/java/frc/robot/Constants.java:120]: I don't think we need most of these constants if we are using the SwerveAPI. Once we get the drive working, we should remove as much of these as possible to keep it slim

**Review by jkleiber — CHANGES_REQUESTED** [review]
@minhnguyenbhs the problem with the robot not moving is likely because RobotContainer is not instantiated in Robot.java. See https://github.com/team401/2023-Robot-Code/blob/main/src/main/java/frc/robot/Robot.java

Specifically we need a private variable `RobotContainer robotContainer;`  and then in robotInit() to initialize with `robotContainer = new RobotContainer()`.

I think we had this is the past, but it must have gotten lost in the rebase.

Additionally, the auto paths shouldn't be in this pull request - that should be a separate branch since it should be a different ticket

File: src/main/java/frc/robot/Constants.java (hunk 3 above)
Comment (minhnguyenbhs) [reply src/main/java/frc/robot/Constants.java:120]: I think that robotContainer is initialized, but at the bottom of robotInit?

Comment (minhnguyenbhs) [reply src/main/java/frc/robot/Constants.java:120]: The code is running, I think- SmartDashboard responds when the joysticks are moved

Comment (jkleiber) [reply src/main/java/frc/robot/Constants.java:120]: Robot.java doesn't show up in this pull request - I would check that this file is correct first

**Review by redPlover — DISMISSED** [review]
DriveWithJoySticks is being weird, I'm going to switch this to a draft

**Comment by redPlover** [comment]
Charlie and I went over the code and found the problem DriveWIthJoysticks; the API was being used incorrectly. Works now 👍

File: src/main/java/frc/robot/subsystems/drive/CommandSwerveDrivetrain.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        super(driveTrainConstants, OdometryUpdateFrequency, modules);
+    }
+
+    public CommandSwerveDrivetrain(SwerveDrivetrainConstants driveTrainConstants, SwerveModuleConstants... modules) {
+        super(driveTrainConstants, modules);
+    }
+
+    public Command applyRequest(Supplier<SwerveRequest> requestSupplier) {
+        return run(() -> setControl(requestSupplier.get()));
+    }
```
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/drive/CommandSwerveDrivetrain.java:32]: I don't think we need this function anymore?

File: src/main/java/frc/robot/commands/DriveWithJoysticks.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            // If not moving, brake
+            drivetrain.setControl(brake);
+        } else if (fieldCentric.getAsBoolean()) {
+            drivetrain.setControl(drive.withVelocityX(-xMpS)
+                    .withVelocityY(-yMpS)
+                    .withRotationalRate(-rotRadpS));
+        } else {
+            drivetrain.setControl(driveRobot.withVelocityX(-xMpS)
+                    .withVelocityY(-yMpS)
+                    .withRotationalRate(-rotRadpS));
```
Comment (jkleiber) [inline src/main/java/frc/robot/commands/DriveWithJoysticks.java:73]: If possible I'd prefer that we put `setControl` calls in `periodic` for the drivetrain and then use one or more setters to set the raw control values in this command

**Review by jkleiber — COMMENTED** [review]
A couple small comments, but looks close to merging

**Review by jkleiber — APPROVED** [review]

**Pull request merged**